echo_server.py

```diff
@@ -50,22 +52,45 @@ def response_error(error_num, error_msg):
 
 def parse_request(request):
     request_list = request.split()
```

Comment:
Because HTTP is a line-based protocol, the usual way of dealing with it is first to split it into individual lines, and then to   parse those lines one at a time.  You only have to deal with the first line, but this is still a better way to do the job than just splitting on whitespace like you do here.  There's valuable information lost later in the request when you do this.